### PR TITLE
chore: Update display, and add initial scenario tests

### PR DESCRIPTION
**Pull request by alamb**
Part of https://github.com/influxdata/idpe/issues/17089

# Rationale
We need a way to see what the compactor will do without having to test it in production

# Changes:
1. Improve the size of the file display per @NGA-TRAN 's suggestion on https://github.com/influxdata/influxdb_iox/pull/6932#discussion_r1101939679
2. Implement the two scenarios @NGA-TRAN  suggested from https://github.com/influxdata/influxdb_iox/pull/6877#issuecomment-1419704850 as tests
3. reduce code replication for layout tests

### CODE REVIEW

File: compactor2/src/tests.rs [hunk 1]
```diff
@@ -605,3 +528,241 @@ async fn assert_skipped_compactions<const N: usize>(
...
+        .simulate_without_object_store()
+        .with_max_desired_file_size_bytes(one_hundred_mb)
+        .build()
+        .await;
+
+    // Model several non overlapping L1 file and new L0 files written
+    // that are not overlapping
+    //
+    // L1: 100MB, 100MB, 100MB, 100MB
+    // L0: 5k, 5k, 5k, 5k, 5k (all non overlapping with the L1 files)
```
Comment (alamb) [inline compactor2/src/tests.rs:637]: Neither this output, nor the one below it seem to produce the output described by @NGA-TRAN  in https://github.com/influxdata/influxdb_iox/pull/6877#issuecomment-1419704850

I suspect there are some other settings that need to be configured

File: compactor2/src/tests.rs [hunk 2]
```diff
@@ -605,3 +528,241 @@ async fn assert_skipped_compactions<const N: usize>(
...
+    - "L1                                                                                                  "
+    - "L1.1[50,100] 20mb   |--L1.1--|                                                                      "
+    - "L1.2[100,150] 50mb            |--L1.2--|                                                            "
+    - "L1.3[150,200] 20mb                      |--L1.3--|                                                  "
+    - "L1.4[200,250] 3mb                                 |--L1.4--|                                        "
+    - "**** Simulation Run 0, type=split(split_times=[421])"
+    - "Input, 7 files: 1, 2, 3, 4, 5, 6, 7"
+    - "**** Output Files"
+    - "L1                                                                                                  "
+    - "L1.8[50,421] 100.17mb|----------------------------------L1.8----------------------------------|      "
```
Comment (alamb) [inline compactor2/src/tests.rs:764]: Likewise this seems to have rewritten all the files for some reason, which I don't understand

File: compactor2/src/tests.rs [hunk 3]
```diff
@@ -605,3 +528,241 @@ async fn assert_skipped_compactions<const N: usize>(
...
+    - "L0.9[500,550] 5kb                                                                           |-L0.9-|"
+    - "L1                                                                                                  "
+    - "L1.1[50,100] 100mb  |-L1.1-|                                                                        "
+    - "L1.2[100,150] 100mb         |-L1.2-|                                                                "
+    - "L1.3[150,200] 100mb                 |-L1.3-|                                                        "
+    - "L1.4[200,250] 100mb                         |-L1.4-|                                                "
+    - "**** Simulation Run 0, type=split(split_times=[175, 300, 425])"
+    - "Input, 9 files: 1, 2, 3, 4, 5, 6, 7, 8, 9"
+    - "**** Output Files"
+    - "L1, all files 100.01mb                                                                              "
```
[the remark this answers is not in the document]
Comment (NGA-TRAN) [reply compactor2/src/tests.rs:688]: Do you turn on `TargetLevel` version in the setup? It looks to me the max_desired file size is set to 100MB but make sure you have it set so we only upgrade large L1 files

File: compactor2/src/tests.rs (hunk 2 above)
Comment (NGA-TRAN) [reply compactor2/src/tests.rs:764]: I think your setup is stillAllAtOnce or/and the max_desired_file_size so not the right value you want

**Comment by alamb** [comment]
> I think your setup is stillAllAtOnce or/and the max_desired_file_size so not the right value you want

Yes, you are right that the algorithm was set incorrectly -- I will fix that

However, when I do that I get another panic that I need to debug:

```

---- tests::layout_l1_with_new_non_overlapping_l0_larger stdout ----
2023-02-10T18:18:41.309333Z  INFO compactor2::components::partitions_source::logging: Fetch partitions n_partitions=1
2023-02-10T18:18:41.309360Z  INFO compactor2::driver: compact partition partition_id=1
2023-02-10T18:18:41.309473Z  INFO compactor2::components::partition_source::logging: Fetch a partition partition_id=1
2023-02-10T18:18:41.309644Z  INFO compactor2::driver: Compaction Plan target_level="CompactionLevel::L1" files_to_compacts=3 files_to_upgrade=0 files_to_keep=4
2023-02-10T18:18:41.309689Z  INFO compactor2::components::ir_planner::logging: created IR plan partition_id=1 n_input_files=3 input_file_size_bytes=15728640 n_output_files=2 compaction_level=1 plan=split
2023-02-10T18:18:41.309714Z  INFO compactor2::driver: job semaphore acquired partition_id=1
2023-02-10T18:18:41.313427Z  INFO compactor2::components::parquet_files_sink::simulator: Simulating split
2023-02-10T18:18:41.313470Z  INFO compactor2::components::parquet_files_sink::simulator: creating output file with fraction of output p=0.8 min_time=Timestamp(300) max_time=Timestamp(420) file_size_bytes=12582912 row_count=2 compaction_level=FileNonOverlapped
2023-02-10T18:18:41.313517Z  INFO compactor2::components::parquet_files_sink::simulator: creating output file with fraction of output p=0.2 min_time=Timestamp(420) max_time=Timestamp(450) file_size_bytes=3145728 row_count=0 compaction_level=FileNonOverlapped
2023-02-10T18:18:41.313570Z  INFO compactor2::driver: job semaphore released partition_id=1
2023-02-10T18:18:41.313783Z  INFO compactor2::components::commit::logging: committed parquet file change target_level=FileNonOverlapped partition_id=1 files_delete=3 files_upgrade=0 files_create=2 bytes_delete=15728640 bytes_upgrade=0 bytes_create=15728640 rows_delete=3 rows_upgrade=0 rows_create=2 delete=[7, 6, 5] upgrade=[] create=[8, 9]
2023-02-10T18:18:41.313903Z  INFO compactor2::driver: Compaction Plan target_level="CompactionLevel::L2" files_to_compacts=6 files_to_upgrade=0 files_to_keep=0
2023-02-10T18:18:41.313949Z  INFO compactor2::components::ir_planner::logging: created IR plan partition_id=1 n_input_files=6 input_file_size_bytes=113246208 n_output_files=2 compaction_level=2 plan=split
2023-02-10T18:18:41.313979Z  INFO compactor2::driver: job semaphore acquired partition_id=1
thread 'tests::layout_l1_with_new_non_overlapping_l0_larger' panicked at 'Sort key cannot be empty', schema/src/sort.rs:251:13
stack backtrace:
   0: rust_begin_unwind
```

Setting back to draft while I work on that

**Comment by alamb** [comment]
I debugged the issue -- and it is that the parquet files that the simulator creates are not sufficiently 'real' enough so the subsequent PR fails. I have some thoughts on how to fix this @NGA-TRAN but I would like to do it as a follow on PR. Is that ok?

**Comment by alamb** [comment]
Thanks for the review @NGA-TRAN

**Pull request merged**